### PR TITLE
feat: implement product diffing for pricing models

### PR DESCRIPTION
**Pull request by agreea**
## What Does this PR Do?

Implements the product diffing functionality (PR 3) from the Pricing Model Diffing gameplan. Adds a `diffProducts` function that compares product arrays and identifies which products need to be removed, created, or updated, including detailed price comparison tracking.

## Summary of Changes

- **New Type**: `ProductDiffInput` - wrapper for `SetupPricingModelProductInput`
- **New Type**: `ProductDiffResult` - diff result with `toRemove`, `toCreate`, `toUpdate` arrays and price diff information
- **New Function**: `diffProducts()` - compares product arrays by `product.slug` and includes price differences in updates
- **Helper Functions**: `getProductSlug()`, `toSluggedProducts()`, and `pricesAreDifferent()` for internal logic
- **Comprehensive Tests**: 11 test cases covering all removal, creation, update, and price comparison scenarios

## Test Coverage

All 28 tests pass (17 existing + 11 new):
- Product removal, creation, and updates
- Price difference detection (unit price, type changes, etc.)
- Edge cases (empty arrays, identical prices, feature changes)
- Property preservation in diff results

---
## Summary by cubic
Implements product diffing for pricing models to detect product creates, updates, and removals, with price change tracking. Adds a diffProducts utility used to compare existing vs. proposed product arrays by slug.

- **New Features**
  - diffProducts compares products by product.slug and returns toRemove, toCreate, and toUpdate; includes priceDiff when prices differ or are missing.
  - Added ProductDiffInput and ProductDiffResult types.
  - Introduced helpers for slug extraction and price comparison; added 11 focused tests (all 28 tests pass).

<sup>Written for commit 3cae8b51c3808177513324aca6349e6928807bdc. Summary will update automatically on new commits.</sup>

## Summary by CodeRabbit

* **New Features**
  * Added product diffing to detect product creations, removals, and updates; surfaces detailed price changes when unit price or price type differ.

* **Tests**
  * Expanded test suite covering creation, removal, updates (name, price/type), mixed and edge cases, preservation of product properties, and ensuring feature-only changes don't produce price diffs.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: platform/flowglad-next/src/utils/pricingModels/diffing.ts [hunk 1]
```diff
@@ -174,3 +184,162 @@ export const diffUsageMeters = (
...
+  if (existingPrice === undefined && proposedPrice === undefined) {
+    return false
+  }
+  // One undefined and one defined means different
+  if (existingPrice === undefined || proposedPrice === undefined) {
+    return true
+  }
+  // Both defined - compare by JSON stringification
+  return (
+    JSON.stringify(existingPrice) !== JSON.stringify(proposedPrice)
```
Comment (angihe93) [inline platform/flowglad-next/src/utils/pricingModels/diffing.ts]: is this robust against object field ordering and other things? maybe isEqual from 'lodash-es' would be better?

Comment (agreea) [reply platform/flowglad-next/src/utils/pricingModels/diffing.ts]: went with `R.equals` which basically does the same thing!
